docs/current/guides/213240-tekton.md

```diff
@@ -0,0 +1,136 @@
+
+```shell
+tkn hub install task git-clone
+```
+
+## Step 3: Create a Tekton Pipeline, Task and PipelineRun
+
+Follow the steps below:
+
+1. Define a new Tekton Pipeline as follows, in a file named `git-pipeline.yaml`.
+
+  ```yaml file=./snippets/tekton/git-pipeline.yaml
+  ```
+
+  This Pipeline references two Tasks:
+
+    - The `git-clone` Task, to check out the Git repository for the project into a Tekton Workspace;
+    - A custom `dagger` Task, to run the Dagger pipeline for the project (defined below).
+
+1. Define a new Tekton Task as follows, in a file named `dagger-task.yaml`.
+
+  ```yaml file=./snippets/tekton/dagger-task.yaml
+  ```
+
+  This Task installs the dependencies needed to execute the Dagger pipeline for the project (which was checked out in the previous Tekton Pipeline) and then executes the pipeline using the `npm run ci` command.
+
+1. Define a new Tekton PipelineRun as follows, in a file named `git-pipeline-run.yaml`.
+
+  ```yaml file=./snippets/tekton/git-pipeline-run.yaml
+  ```
+
+  This PipelineRun corresponds to the Tekton Pipeline created previously. It executes the Tekton Pipeline with a given set of input parameters: the Git repository URL and an optional Dagger Cloud token. Update the `YOUR_REPOSITORY_URL` placeholder with the correct repository URL and, if you have a Dagger Cloud token, replace the `YOUR_DAGGER_CLOUD_TOKEN` placeholder with the token or a Kubernetes secret holding the token.
+
+A few important points to note:
+
+- In the Task, the Dagger Engine runs as a sidecar and shares a socket with the Task itself. The Task uses `dind` as its runtime in order to have Docker available.
+- Setting the `DAGGER_CLOUD_TOKEN` environment variable in the Task and the `dagger-cloud-token` parameter in the Pipeline and PipelineRun is only necessary if integrating with [Dagger Cloud](https://dagger.cloud/).
+
+## Step 4: Run the Pipeline
+
+1. Apply the configuration:
+
+```shell
+kubectl apply -f dagger-task.yaml
+kubectl apply -f git-pipeline-yaml
+```
+
+1. Run the Tekton Pipeline:
+
+```shell
+kubectl create -f git-pipeline-run.yaml
+```
+
+To see the logs from the PipelineRun, obtain the pod name and in another terminal run `kubectl logs -f POD_NAME`
+
+Once the PipelineRun has successfully completed, run it again. Dagger's caching should result in a significantly faster second execution.
+
+## Conclusion
+
+This example demonstrated how to integrate Dagger into a Tekton Pipeline using a custom Task.
+
+To learn more about Dagger, use the [API Key Concepts](../api/975146-concepts.mdx) page and the [Go](https://pkg.go.dev/dagger.io/dagger), [Node.js](../sdk/nodejs/reference/modules.md) and [Python](https://dagger-io.readthedocs.org/) SDK References. For more information on Tekton, refer to the [official documentation](https://tekton.dev/docs/).
+
+## Appendix A: Create a GitHub repository with an example project
```

Comment:
@kpenfound I modified your initial draft to instead suggest using the hello-dagger starter app. I have also made some changes to the setup script in the Task. However I couldn't test it properly as I hit the following error, both with your original version and my modifications:

```
file:///workspace/source/node_modules/@dagger.io/dagger/dist/provisioning/bin.js:178
                        reject(new EngineSessionConnectionTimeoutError("Engine connection timeout", { timeOutDuration }));
                               ^

EngineSessionConnectionTimeoutError: Engine connection timeout
    at Timeout._onTimeout (file:///workspace/source/node_modules/@dagger.io/dagger/dist/provisioning/bin.js:178:32)
    at listOnTimeout (node:internal/timers:569:17)
    at process.processTimers (node:internal/timers:512:7) {
  cause: undefined,
  code: 'D104',
  timeOutDuration: 300000
}
```

I suspect this may be because I'm running Tekton on minikube and hitting a resource limit on the host. Could you please re-test once and modify as needed?